### ISSUE TITLE
no-extra-parens false positive for ctor

### ISSUE THREAD

**Issue by orgads** [eslint/eslint#13504]
**Tell us about your environment**

* **ESLint Version:** v7.4.0
* **Node Version:** v12.18.2
* **npm Version:** v6.14.5

**What parser (default, Babel-ESLint, etc.) are you using?**
@typescript-eslint/parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  "env": {
    "es6": true,
    "node": true
  },
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking"
  ],
  "globals": {
    "globalThis": false, // means it is not writeable
  },
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2018,
    "project": "./tsconfig.json",
    "sourceType": "module"
  },
  "plugins": [
    "@typescript-eslint"
  ],
  "reportUnusedDisableDirectives": true,
  "rules": {
    "no-extra-parens": ["error", "all", { "nestedBinaryExpressions": false }]
  }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
class Foo
{
}

const foo = new Map<string, new() => Foo>([
  ['foo', Foo]
]);
```

```bash
eslint --ext .ts .
```

**What did you expect to happen?**
It should not report unnecessary parentheses, as they are necessary.

**What actually happened? Please include the actual, raw output from ESLint.**
It reports no-extra-parens (for the parentheses after `=> Foo>`)

**Are you willing to submit a pull request to fix this bug?**
Sorry, but I don't know how to fix this.

**Comment by anikethsaha** [comment]
Thanks for the report 👍 

You are using `Typescript` and eslint doesn't support TS by default. You need to use the [`@typescript-eslint/no-extra-parens`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-extra-parens.md) instead of eslint's `no-extra-parens`. 

This is working fine  for non [TS codes](https://eslint.org/demo#eyJ0ZXh0IjoiY29uc3QgZm9vID0gbmV3IE1hcChbXG4gIFx0Wydmb28nLCBGb29dXG5dKTsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjExLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImluZGVudCI6Miwibm8tZXh0cmEtcGFyZW5zIjoyfSwiZW52Ijp7ImJyb3dzZXIiOnRydWUsIm5vZGUiOnRydWUsImNvbW1vbmpzIjp0cnVlLCJzaGFyZWQtbm9kZS1icm93c2VyIjp0cnVlLCJ3b3JrZXIiOnRydWUsImFtZCI6dHJ1ZSwibW9jaGEiOnRydWUsImphc21pbmUiOnRydWUsImplc3QiOnRydWUsInBoYW50b21qcyI6dHJ1ZSwianF1ZXJ5Ijp0cnVlLCJxdW5pdCI6dHJ1ZSwicHJvdG90eXBlanMiOnRydWUsInNoZWxsanMiOnRydWUsIm1ldGVvciI6dHJ1ZSwibW9uZ28iOnRydWUsInByb3RyYWN0b3IiOnRydWUsImFwcGxlc2NyaXB0Ijp0cnVlLCJuYXNob3JuIjp0cnVlLCJzZXJ2aWNld29ya2VyIjp0cnVlLCJhdG9tdGVzdCI6dHJ1ZSwiZW1iZXJ0ZXN0Ijp0cnVlLCJ3ZWJleHRlbnNpb25zIjp0cnVlLCJlczYiOnRydWUsImVzMjAxNyI6dHJ1ZSwiZXMyMDIwIjp0cnVlLCJncmVhc2Vtb25rZXkiOnRydWV9fX0=)

**Comment by orgads** [comment]
I just tried with `@typescript-eslint/no-extra-parens`, and it reports the same error.

**Comment by orgads** [comment]
Found https://github.com/typescript-eslint/typescript-eslint/issues/1587

I'll close this one.
